packages/@aws-cdk/pipelines/lib/pipeline.ts

```diff
@@ -157,13 +167,20 @@ export class CdkPipeline extends CoreConstruct {
       if (props.crossAccountKeys !== undefined) {
         throw new Error('Cannot set \'crossAccountKeys\' if an existing CodePipeline is given using \'codePipeline\'');
       }
+      if (props.enableKeyRotation !== undefined) {
+        throw new Error('Cannot set \'enableKeyRotation\' if an existing CodePipeline is given using \'codePipeline\'');
+      }
+      if (props.enableKeyRotation && !props.crossAccountKeys) {
+        throw new Error('When setting \'enableKeyRotation\' you must also set \'crossAccountKeys\'');
+      }
 
       this._pipeline = props.codePipeline;
     } else {
       this._pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
         pipelineName: props.pipelineName,
         crossAccountKeys: props.crossAccountKeys,
         restartExecutionOnUpdate: true,
+        enableKeyRotation: props.enableKeyRotation,
```

Comment:
Maybe move this above `restartExecutionOnUpdate`, to the `crossAccountKeys: props.crossAccountKeys,` line? They seem very related.